stemcell_builder/stages/system_azure_wala/apply.sh

```diff
@@ -26,22 +27,18 @@ run_in_chroot $chroot "
   tar zxvf wala.tar.gz
   cd WALinuxAgent-${wala_release}
   sudo python setup.py install --skip-data-files
-  cp bin/waagent /usr/sbin/waagent
+  cp bin/* /usr/sbin/
   chmod 0755 /usr/sbin/waagent
+  chmod 0755 /usr/sbin/waagent2.0
   cd /tmp/
   sudo rm -fr WALinuxAgent-${wala_release}
   rm wala.tar.gz
 "
-
 cp -f $dir/assets/etc/waagent.conf $chroot/etc/waagent.conf
+cp -f $dir/assets/etc/walinuxagent.service $chroot/lib/systemd/system/walinuxagent.service
+chmod 0755 $chroot/lib/systemd/system/walinuxagent.service
 
-cp -a $dir/assets/runit/waagent $chroot/etc/sv/waagent
-
-# Set up waagent with runit
-run_in_chroot $chroot "
-chmod +x /etc/sv/waagent/run
-ln -s /etc/sv/waagent /etc/service/waagent
-"
+run_in_bosh_chroot $chroot "systemctl enable walinuxagent.service"
```

Comment:
This doesn't need to use `run_in_bosh_chroot`. Should use `run_in_chroot` instead.

Reply:
resolved.